File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountMatchedPrefixWord/CountMatchedPrefixWord_test.js
import assert from 'assert'
import solution from './CountMatchedPrefixWord.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const array = ['apple', 'banana', 'kakao', 'naver', 'apache']
    const p = 'a'

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const array = ['asjefkl', 'aeklfj', 'asifljl', 'ruihgjd', 'djgkln']
    const p = 'as'

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const array = ['cloff', 'vxqql', 'tkafv', 'ophik', 'lscmj']
    const p = 't'

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const array = []
    const p = 'naver'

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const array = ['uehex', 'gevhl', 'bhitm', 'csoqm', 'xsqcf']
    const p = 'naver'

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const array = ['mmzuf', 'sfnki', 'qdpov', 'zpdtg', 'prozo']
    const p = ''

    // when
    const result = solution(array, p)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });
})

File: 4기 코딩테스트_220615/4기 - 2nd 코딩테스트/CountMatchedPrefixWord/CountMatchedPrefixWord_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.CountMatchedPrefixWord.CountMatchedPrefixWord import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        array = ['apple', 'banana', 'kakao', 'naver', 'apache']
        p = 'a'

        # when
        result = solution(array, p)

        # then
        self.assertEqual(2, result)

    def test_2(self):
        # given
        array = ['asjefkl', 'aeklfj', 'asifljl', 'ruihgjd', 'djgkln']
        p = 'as'

        # when
        result = solution(array, p)

        # then
        self.assertEqual(2, result)

    def test_3(self):
        # given
        array = ['cloff', 'vxqql', 'tkafv', 'ophik', 'lscmj']
        p = 't'

        # when
        result = solution(array, p)

        # then
        self.assertEqual(1, result)

    def test_4(self):
        # given
        array = []
        p = 'naver'

        # when
        result = solution(array, p)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        array = ['uehex', 'gevhl', 'bhitm', 'csoqm', 'xsqcf']
        p = 'naver'

        # when
        result = solution(array, p)

        # then
        self.assertEqual(0, result)

    def test_6(self):
        # given
        array = ['mmzuf', 'sfnki', 'qdpov', 'zpdtg', 'prozo']
        p = ''

        # when
        result = solution(array, p)

        # then
        self.assertEqual(5, result)


if __name__ == '__main__':
    unittest.main()
